misc/python/materialize/checks/identifiers.py

```diff
@@ -0,0 +1,615 @@
+        'Kernel.exec("ls -al /")',
+        "Kernel.exit(1)",
+        "%x('ls -al /')",
+        '<?xml version="1.0" encoding="ISO-8859-1"?><!DOCTYPE foo [ <!ELEMENT foo ANY ><!ENTITY xxe SYSTEM "file:///etc/passwd" >]><foo>&xxe;</foo>',
+        "$HOME",
+        "$ENV{'HOME'}",
+        "%d",
+        "%s%s%s%s%s",
+        "{0}",
+        "%*.*s",
+        "%@",
+        "%n",
+        "File:///",
+        "../../../../../../../../../../../etc/passwd%00",
+        "../../../../../../../../../../../etc/hosts",
+        "() { 0; }; touch /tmp/blns.shellshock1.fail;",
+        "() { _; } >_[$($())] { touch /tmp/blns.shellshock2.fail; }",
+        "<<< %s(un='%s') = %u",
+        "+++ATH0",
+        "CON",
+        "PRN",
+        "AUX",
+        "CLOCK$",
+        "NUL",
+        "A:",
+        "ZZ:",
+        "COM1",
+        "LPT1",
+        "LPT2",
+        "LPT3",
+        "COM2",
+        "COM3",
+        "COM4",
+        "DCC SEND STARTKEYLOGGER 0 0 0",
+        "Scunthorpe General Hospital",
+        "Penistone Community Church",
+        "Lightwater Country Park",
+        "Jimmy Clitheroe",
+        "Horniman Museum",
+        "shitake mushrooms",
+        "RomansInSussex.co.uk",
+        "http://www.cum.qc.ca/",
+        "Craig Cockburn, Software Specialist",
+        "Linda Callahan",
+        "Dr. Herman I. Libshitz",
+        "magna cum laude",
+        "Super Bowl XXX",
+        "medieval erection of parapets",
+        "evaluate",
+        "mocha",
+        "expression",
+        "Arsenal canal",
+        "classic",
+        "Tyson Gay",
+        "Dick Van Dyke",
+        "basement",
+        "If you're reading this, you've been in a coma for almost 20 years now. We're trying a new technique. We don't know where this message will end up in your dream, but we hope it works. Please wake up, we miss you.",
+        "Roses are \u001b[0;31mred\u001b[0m, violets are \u001b[0;34mblue. Hope you enjoy terminal hue",
+        "But now...\u001b[20Cfor my greatest trick...\u001b[8m",
+        "The quic\b\b\b\b\b\bk brown fo\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007\u0007x... [Beeeep]",
+        "Powerلُلُصّبُلُلصّبُررً ॣ ॣh ॣ ॣ冗",
+        "🏳0🌈️",
+        "జ్ఞ‌ా",
+        "گچپژ",
+        "{% print 'x' * 64 * 1024**3 %}",
+        '{{ "".__class__.__mro__[2].__subclasses__()[40]("/etc/passwd").read() }}',
+    ]
+
+    def initialize(self) -> Testdrive:
+        cmds = "\n".join(
+            [
+                f"""
+            > CREATE DATABASE {dq(ident + "_db")};
+            > SET DATABASE={dq(ident + "_db")};
+            > CREATE SCHEMA {dq(ident + "_schema")};
+            > CREATE TYPE {dq(ident + "_type")} AS LIST (ELEMENT TYPE = text);
+            > CREATE TABLE {dq(ident + "_schema")}.{dq(ident + "_table")} ({dq(ident)} TEXT, c2 {dq(ident + "_type")});
+            > INSERT INTO {dq(ident + "_schema")}.{dq(ident + "_table")} VALUES ({sq(ident)}, LIST[{sq([ident])}]::{dq(ident + "_type")});
```

Comment:
Please do `CREATE MATERIALIZED VIEW` over the table so that the `clusterd` is also involved . The `SELECT` portion of the view must be more complex than `*` so that it does not get inlined now or in the future, e.g. `SELECT COUNT(col)` would be good and type-insensitive.

`CREATE SOURCE` would also be a good thing to have.